### PR TITLE
HBASE-23651 Region balance throttling can be disabled

### PR DESCRIPTION
**Pull request by binlijin**

### CODE REVIEW

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java [hunk 1]
```diff
@@ -1860,11 +1856,14 @@ public boolean balance(boolean force) throws IOException {
         //rpCount records balance plans processed, does not care if a plan succeeds
         rpCount++;
 
-        balanceThrottling(balanceStartTime + rpCount * balanceInterval, maxRegionsInTransition,
+        if (this.maxBlancingTime > 0) {
```
Comment (virajjasani) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java]: ok so by disabling throttling what we mean is providing negative int value for config `hbase.balancer.max.balancing` in master site config

Comment (binlijin) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java]: Yes.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 56s |  master passed  |
| +1 :green_heart: |  compile  |   0m 58s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 51s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 43s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 40s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  7s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 28s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 44s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m  6s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 47s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 158m 47s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   | 218m 59s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.assignment.TestRegionMoveAndAbandon |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-991/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/991 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux e039be0b793b 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-991/out/precommit/personality/provided.sh |
| git revision | master / 33f45d4413 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-991/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-991/1/testReport/ |
| Max. process+thread count | 4983 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-991/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java [hunk 2]
```diff
@@ -1688,12 +1688,8 @@ InetAddress getRemoteInetAddress(final int port,
    * @return Maximum time we should run balancer for
    */
   private int getMaxBalancingTime() {
-    int maxBalancingTime = getConfiguration().getInt(HConstants.HBASE_BALANCER_MAX_BALANCING, -1);
-    if (maxBalancingTime == -1) {
-      // if max balancing time isn't set, defaulting it to period time
-      maxBalancingTime = getConfiguration().getInt(HConstants.HBASE_BALANCER_PERIOD,
-        HConstants.DEFAULT_HBASE_BALANCER_PERIOD);
-    }
+    int maxBalancingTime = getConfiguration()
```
Comment (anoopsjohn) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java]: This change is wrong right?  We no longer check any configured value for HBASE_BALANCER_PERIOD.
U wanted it like 
maxBalancingTime = getConfiguration().getInt(HConstants.HBASE_BALANCER_MAX_BALANCING,  getConfiguration().getInt(HConstants.HBASE_BALANCER_PERIOD,
        HConstants.DEFAULT_HBASE_BALANCER_PERIOD);

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java (hunk 1 above)
Comment (anoopsjohn) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java]: So you will change this maxBlancingTime dynamically at HM config?  
If permanently for the cluster the throttling needs disabled, hbase.master.balancer.maxRitPercent can be 100% right (which is the def value).  What am missing here? I did not read this part of code fully though

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java (hunk 2 above)
Comment (binlijin) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java]: Yes, you are right.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java (hunk 1 above)
Comment (binlijin) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java]: No this do not change maxBlancingTime dynamically, It just skip the balanceThrottling when maxBlancingTime<=0.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  3s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 33s |  master passed  |
| +1 :green_heart: |  compile  |   0m 56s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 24s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 38s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 30s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 28s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 39s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 53s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 48s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 155m 29s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 34s |  The patch does not generate ASF License warnings.  |
|  |   | 215m 24s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-991/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/991 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux f474e9787b27 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-991/out/precommit/personality/provided.sh |
| git revision | master / 33f45d4413 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-991/2/testReport/ |
| Max. process+thread count | 5043 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-991/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by binlijin** [comment]
@virajjasani @anoopsjohn  Thanks very much for the review.

**Pull request merged**